### PR TITLE
enable smtps identified from the server URI

### LINKED ISSUE #2568: Allow SSL over SMTP and not only STARTTLS
Hello

I configured my SMTP server to disable STARTTLS and only allow SSL over SMTP (465).
This kind of configuration is not uncommon I think, however I couldn't find a way to implement it.

In the end, I had to patch the lib/smtpserver.py by replacing (line 107) : 

> mail = smtplib.**SMTP**(config['server'], port=int(config['port']),
>                             timeout=config.get('timeout', TIMEOUT))

with

> mail = smtplib.**SMTP_SSL**(config['server'], port=int(config['port']),
>                             timeout=config.get('timeout', TIMEOUT))

It's working fine for me right now but I'm wondering if it would be a good feature to have another checkbox with "SSL" near the "TLS" checkbox in the SMTP server configuration.

I'm not a python guy so I cant' propose a patch for the whole functionnaly

### PR DESCRIPTION
**Pull request by laclaro**
closes #2568

I do not like this if elif else clause - any thoughts on that?

### CODE REVIEW

File: privacyidea/lib/smtpserver.py [hunk 1]
```diff
@@ -103,9 +103,15 @@ def test_email(config, recipient, subject, body, sender=None,
         msg['To'] = ",".join(recipient)
         msg['Date'] = strftime("%a, %d %b %Y %H:%M:%S +0000", gmtime())
         msg['Reply-To'] = reply_to
-
-        mail = smtplib.SMTP(config['server'], port=int(config['port']),
-                            timeout=config.get('timeout', TIMEOUT))
+        if config['server'].startswith('smtps://'):
+            mail = smtplib.SMTP_SSL(config['server'][8:], port=int(config['port']),
```
Comment (cornelinux) [inline privacyidea/lib/smtpserver.py:107]: Please add a test case (mock) for this!
Is it necessary to add some docs?

Have you checked if `SMTP_SSL` exists in Python 2.7.5?

**Review by cornelinux — CHANGES_REQUESTED** [review]

File: privacyidea/lib/smtpserver.py (hunk 1 above)
Comment (laclaro) [reply privacyidea/lib/smtpserver.py:107]: Yes it is also in 2.7.

File: privacyidea/lib/smtpserver.py [hunk 2]
```diff
@@ -103,15 +103,21 @@ def test_email(config, recipient, subject, body, sender=None,
         msg['To'] = ",".join(recipient)
         msg['Date'] = strftime("%a, %d %b %Y %H:%M:%S +0000", gmtime())
         msg['Reply-To'] = reply_to
-
-        mail = smtplib.SMTP(config['server'], port=int(config['port']),
-                            timeout=config.get('timeout', TIMEOUT))
+        if config['server'].lower().startswith('smtps://'):
```
Comment (plettich) [inline privacyidea/lib/smtpserver.py:106]: We should use the builtin `urllib` for that. I'll add the changes.

**Review by plettich — CHANGES_REQUESTED** [review]

File: privacyidea/lib/smtpserver.py (hunk 2 above)
Comment (plettich) [reply privacyidea/lib/smtpserver.py]: Done in 322d2707

**Review by cornelinux — APPROVED** [review]

**Comment by cornelinux** [comment]
HI @plettich  I think you also need to approve to merge this :-)

**Review by plettich — APPROVED** [review]

**Pull request merged**